feature_finder/find_features.py

```
@@ -0,0 +1,106 @@
+                                  false_alarm)
+
+
+MODELS = {
+    'linear': {'model': LinearRegression, 'error': root_mean_squared_error, 'best_selector': 0},
+    'logistic': {'model': LogisticRegression, 'error': accuracy, 'best_selector': -1},
+}
+
+
+class Model:
+    test_size = 0.2
+    k_fold = 5
+
+    def __init__(self, model_type, plugins=None):
+        if model_type not in MODELS:
+            raise TypeError(
+                "Please specify one of the following model types: {}."
+                .format(', '.join(MODELS.keys())))
+        self.model = MODELS[model_type]['model']()
+        self.error = MODELS[model_type]['error']
+        self.plugins = plugins or []
+
+    def select(self, data, y_column):
+        """Get the result of fitting the data on selectors."""
+        if self.plugins:
+            data = self.customize(data)
+
+        train, test = train_test_split(data, test_size=self.test_size)
+        x_train = train.drop(y_column, axis=1)
+        y_train = train[y_column]
+        x_test = test.drop(y_column, axis=1)
+        y_test = test[y_column]
+
+        return {c: self.get_feature_error(c, x_train, y_train, x_test, y_test)
+                for c in x_train.columns}
+
+    def get_feature_error(self, c, x_train, y_train, x_test, y_test):
+        self.model.fit(np.array(x_train[c]).reshape(-1, 1), y_train)
+        predicted = self.model.predict(np.array(x_test[c]).reshape(-1, 1))
+        return self.error(y_test, predicted)
+
+    def customize(self, data):
+        """Apply plugins."""
+        plugins_functions = [
+            obj for name, obj in inspect.getmembers(plugins)
+            if inspect.isfunction(obj) and name in self.plugins
+        ]
+        for fn in plugins_functions:
+            try:
+                data = fn(data)
+            except Exception:
+                raise plugins.PluginException(
+                    'Error with plugin "{}".'.format(fn.__name__))
```

You're hiding the underlying error message -- potentially annoying?

---

yes definitely. ideally i'd create a "verbose" flag that prints out the traceback. maybe in the future.